src/wfs/src/main/java/org/geoserver/wfs/Transaction.java

```diff
@@ -428,6 +419,24 @@ protected TransactionResponse execute(TransactionRequest request) throws Excepti
         // response = build;
     }
 
+    /**
+     * @param request
+     * @param multiplexer
+     * @param stores
+     * @param elementHandlers
+     * @param result
+     * @return a new {@link BatchManager} batching INSERT and DELETE operations where possible.
+     */
+    protected BatchManager createBatchManager(
+            TransactionRequest request,
+            TransactionListenerMux multiplexer,
+            Map<QName, FeatureStore> stores,
+            Map<TransactionElement, TransactionElementHandler> elementHandlers,
+            TransactionResponse result) {
+        return new BatchManager(
```

Comment:
This was done by intention, because it enables replacing the BatchManager implementation without overriding the entire execute() method, which is very long.
Ok for you?

Reply:
Not really. Transaction was not meant for subclassing, I don't see usage in the mocks either. I've now noticed you also made BatchManager protected. Again, why? Do you have specific needs in a fork of GeoServer?

Reply:
Both is to enable replacing BatchManager with a "No-Op" implementation, if required for TransactionListeners etc, which are not happy about request restructuring. 

Currently this is not required for us, but I am just in the middle of upgrading GeoServer, so I am not 100% sure yet.

I'd suggest to keep the API open. Subclassers are operating on their own risk anyway, but it is making life much easier for adoptions.

Reply:
Replacing the usage of Transaction with a subclass is not all that easy, one needs to either modify or replace also DefaultWebFeatureService and DefaultWebFeatureService20. But ok, let's keep it. Since you are making it API, give it readable javadoc then.